### PR TITLE
Add serde support to document type.

### PR DESCRIPTION
**Pull request by thomas-k-cameron**
## Description
Implements serde support for `Document`.

## Testing
Test checks whether the serialized/de-serialized data matches with the expected value

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by thomas-k-cameron** [comment]
I'm closing this because I found it difficult to separate into smaller PR.

**Comment by thomas-k-cameron** [comment]
> It looks like after the smaller prerequisite PRs are merged, the only functionality that still needs to be added is ser/de support for `Document`. Is that correct? If so, could you either repurpose this PR by updating the description/title, or create a new one just for `Document`?

Yes, document type is the only thing that is left.

I changed the title.

File: rust-runtime/aws-smithy-types/src/document.rs [hunk 1]
```diff
@@ -207,3 +228,60 @@ impl From<Number> for Document {
...
+#[cfg(test)]
+mod test {
+    /// checks if a) serialization of json suceeds and b) it is compatible with serde_json
+    #[test]
+    #[cfg(all(
+        aws_sdk_unstable,
+        feature = "serde-serialize",
+        feature = "serde-deserialize"
+    ))]
+    fn serialize_json() {
```
Comment (jdisanti) [inline rust-runtime/aws-smithy-types/src/document.rs:243]: Nice test!

**Review by jdisanti — APPROVED** [review]
Looks good. Thank you!

File: rust-runtime/aws-smithy-types/src/document.rs (hunk 1 above)
Comment (thomas-k-cameron) [reply rust-runtime/aws-smithy-types/src/document.rs:243]: :D

**Pull request merged**